drf_link_navigation_pagination/__init__.py

```diff
@@ -32,6 +34,22 @@ class LinkNavigationPagination(LimitOffsetPagination):
         if hasattr(settings, "DRF_LNP_HEADER_NUMBER_OF_OVERLAP_PATHS")
         else "X-Drf-Number-Overlap-Paths"
     )
+    header_max_pagination_size = (
+        settings.DRF_MAX_PAGINATION_SIZE
+        if hasattr(settings, "DRF_MAX_PAGINATION_SIZE")
+        else "X-Drf-Max-Pagination-Size"
+    )
+
+    def paginate_queryset(self, queryset, request, view=None):
+        if request.headers.get(self.header_max_pagination_size) is not None:
+            # https://www.django-rest-framework.org/api-guide/pagination/#limitoffsetpagination
+            self.max_limit = int(request.headers.get(self.header_max_pagination_size))
+            try:
+                if int(request.query_params.get(self.limit_query_param, 0)) > self.max_limit:
+                    raise BadLimitValue()
+            except ValueError:
+                pass
```

Comment:
**suggestion (non-blocking):** Since we are doing nothing in the `except ValueError` block, we can use  the [suppress](https://docs.python.org/3.7/library/contextlib.html#contextlib.suppress) decorator:

```suggestion
            with suppress(ValueError):
                if int(request.query_params.get(self.limit_query_param, 0)) > self.max_limit:
                    raise BadLimitValue()
```